Lib/os.py

```diff
@@ -651,6 +651,29 @@ def get_exec_path(env=None):
     return path_list.split(pathsep)
 
 
+def get_current_dir_name():
+    """
+    Return a string representing the current working directory taking into
+    consideration the users *PWD* environment variable if it exists. This
+    is opposed to getcwd() which dereferences symlinks in the path. This
+    function is identical to getcwd() on systems that do *not* support
+    the *PWD* environment variable.
+    """
+    cwd = getcwd()
+
+    try:
+        pwd = environ["PWD"]
+    except KeyError:
+        return cwd
+
+    cwd_stat, pwd_stat = map(stat, [cwd, pwd])
+
+    if (cwd_stat.st_dev == pwd_stat.st_dev and
```

Comment:
So, confusion #2 -, assuming that get_current_dir_name() is supported by the platform why is that not just being called rather than rewriting it? I admit my confusion comes from the lack of platform documentation, but why so hard? Wouldn't calling the "platform" implementation be more efficient than writing this in "pure" python.

More simply, why not use the platform implementation, perhaps adding it to posixmodule.c rather than here?